packages/extension/src/ui/features/accountTokens/tokens.state.ts

```diff
@@ -337,17 +338,28 @@ async function fetchFungibleTokenFromFullNode(network: Network, tokenId: string)
       return undefined
     }
 
-    const metadata = await fetchImmutable(`${tokenId}-token-metadata`, () => nodeProvider.fetchFungibleTokenMetaData(tokenId))
-    const token: Token = {
-      id: tokenId,
-      networkId: network.id,
-      name: Buffer.from(metadata.name, 'hex').toString('utf8'),
-      symbol: Buffer.from(metadata.symbol, 'hex').toString('utf8'),
-      decimals: metadata.decimals,
-      verified: false
+    const metadata = await fetchImmutable(`${tokenId}-token-metadata`, async () => {
+      try {
+        return (await nodeProvider.fetchFungibleTokenMetaData(tokenId))
+      } catch (e: any) {
+        if (e.message.startsWith('Failed to call contract')) {
+          return { name: undefined, symbol: undefined, decimals: 0 }
```

Comment:
[Nit]: Perhaps return `undefined` directly.

Reply:
If we return `undefined` it would be the same as not cached ([here](https://github.com/alephium/extension-wallet/blob/master/packages/extension/src/shared/utils/fetchImmutable.ts#L11-L12)), this way it can be cached so it will not make additional calls.

Reply:
Got it, thanks for the explanation